mirest/ah-haven-space-sprinters
# #164694221 Authors should be able to create articles

**Description of tasks to be completed**

Have the following endpoints working

`POST api/articles`
`GET api/articles`
`GET api/articles/<slug>`
`PUT api/articles/<slug>`
`DELETE api/articles/<slug>`

**How should this be manually tested**

- clone the repo and cd into the directory then create a virtualenv and activate 
- install all the requirements using command 'pip install -r requirements.txt'
- ensure to set all the environment variables in the .env file and the run the server using command 'python manage.py runserver'
- open postman or your favourite client and test out all the endpoints listed above
- use this data format in the body {"article":{"title":"title","body":"body","description":"description"}}

Alternatively 
- You can test out the above endpoints using a link provided by the review app with the same data format

**What are the relevant pivotal tracker stories**

#164694221

**Screenshots**

- successfully created article
![create article](https://user-images.githubusercontent.com/40593659/55514364-8dfced80-5670-11e9-9e37-2a7e29389633.PNG)

- view all articles 
![multiple art](https://user-images.githubusercontent.com/40593659/55514456-d3211f80-5670-11e9-8272-953332413909.PNG)

- article updated successfully
![update art](https://user-images.githubusercontent.com/40593659/55514544-17acbb00-5671-11e9-81c3-972218c3019c.PNG)

- artical successfully deleted
![del art](https://user-images.githubusercontent.com/40593659/55514313-6f96f200-5670-11e9-87b7-c6f0b5c5a22c.PNG)
